### PR TITLE
cans spawn sealed

### LINKED ISSUE #40127: Canned items are spawning opened again
# Describe the bug

Canned items are spawning opened rather than sealed. 

# Steps To Reproduce

1. Create a new world. Spawn a character.
2a. Teleport around until you find somewhere with canned goods. (I looked at both an evac shelter and a house.)
2b. Alternatively, use debug spawn to spawn a canned item. The can spawns opened.
3. Optionally, accelerate time to confirm that the items do rot.

# Expected behavior

Canned goods should spawn sealed, rather than opened as if some passive aggressive survivor had been through the area prior.
This bug also prevents storing canned goods in clothes, since they must be sealed for that to be possible.

# Screenshots

image
image after 20 days 
image

# Versions and configuration

OS: Windows 10
Version: 0.E-1713-gd15c896
Graphics: Tiles
Language: English
Mods: 
image

# Additional context

None

### LINKED ISSUE #40658: Cannot stash cans
# Describe the bug

When attempting to pick up cans, the following message appears:
"Can't stash aluminum can of beer while it's not empty". 
The can in question shouldn't be open. 

# Steps To Reproduce

1. Roll a new character.
2. Loot a house's fridge.
3. Try to grab a can out of the house's fridge. 

# Expected behavior

I expect to be able to grab the can out of the fridge and place it in my backpack by using "e" to examine the fridge, pressing the hotkey for the can and then hitting enter.

# Screenshots

CanBug1
CanBug2

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.E-2203-g30952d6 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    C.R.I.T Expansion Mod [crt_expansion],
    Crazy Cataclysm [crazy_cataclysm],
    Cars to Wrecks [cars_to_wrecks],
    No Fungal Monsters [No_Fungi],
    Stats Through Skills [StatsThroughSkills]
]

# Additional context

Adding a save file with the can of beer in the fridge for your convenience
[Lawton.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4641101/Lawton.zip)

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: None```
No need to hit the 0.F changelog but this is a pretty significant bugfix for those who play experimental.
#### Purpose of change
Fixes #40127
Fixes #40658

#### Describe the solution
When the item is being modified in item_group.cpp, make sure to seal it when adding things to the CONTAINER pocket

#### Testing
wander around a town and look for aluminum cans of beer and small tin cans of cat food and mashed pumpkins

#### Additional context

image

Also, i left in the minor code changes i made to help me find the bug. they are minor organizational things but make it a lot easier to put breakpoints in.

### CODE REVIEW

**Comment by Stephen2** [comment]
Should we be adding or modifying a test for this, to ensure it never comes back?

**Comment by Stephen2** [comment]
(Just learning the policies of this repository, so please excuse my question if wrong)

**Comment by anothersimulacrum** [comment]
It'd be great to have a test for this, if you're volunteering to write it :)!

File: src/item_pocket.cpp [hunk 1]
```diff
@@ -596,7 +596,9 @@ bool item_pocket::seal()
 
 void item_pocket::unseal()
 {
-    _sealed = false;
+    if( _sealed ) {
+        _sealed = false;
```
Comment (BevapDin) [inline src/item_pocket.cpp:600]: What's the point of this change? The behaviour seems the be exactly the same as before.

Comment (KorGgenT) [reply src/item_pocket.cpp:600]: i'm not exactly married to keeping it, but my logic was that it made setting breakpoints at that part much easier if you only want to find out when it actually matters that it sets it false

Comment (BevapDin) [reply src/item_pocket.cpp:600]: Set a conditional breakpoint then.

**Comment by Stephen2** [comment]
@anothersimulacrum I'm gonna try and figure out how to write C++ so I can contribute here.  Kind of daunting to jump into :p

File: src/item_pocket.cpp [hunk 2]
```diff
@@ -596,7 +596,9 @@ bool item_pocket::seal()
...
+    }
```
Comment (kevingranade) [inline src/item_pocket.cpp:601]:
```suggestion
    _sealed = false;
```

**Pull request merged**